Services/BrokerService/Commands/GetInstrumentsCommand.cs

```diff
@@ -0,0 +1,36 @@
+﻿using BrokerService.Interfaces;
+using System.Collections.Generic;
+using System;
+using BrokerService.Utils;
+using Kernel.CustomExceptions;
+using DTO;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+namespace BrokerService.Commands
+{
+    public class GetInstrumentsCommand : IGetInstrumentsCommand
+    {
+        private ILogger<GetInstrumentsCommand> logger;
+
+        public GetInstrumentsCommand([FromServices] ILogger<GetInstrumentsCommand> logger)
+        {
+            this.logger = logger;
+        }
+
+        public IEnumerable<Instrument> Execute(BankType bank, string token, int depth, string instrument)
+        {
+            try
+            {
+                logger.LogInformation($"Received request for getting {instrument} from {bank} with given depth {depth}.");
```

Comment:
> You always receive request so you can log this info outside try catch block

What's the difference?

Reply:
Because you will not log that you received request. You will catch only exception but the last positive log message would be from previous step

Reply:
> Because you will not log that you received request. You will catch only exception but the last positive log message would be from previous step

It will enter try block anyway, why would it miss log?